api_shield_api_discovery.go

```diff
@@ -0,0 +1,174 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/goccy/go-json"
+)
+
+// APIShieldDiscoveryOperation is an operation that was discovered by API Discovery.
+type APIShieldDiscoveryOperation struct {
+	// ID represents the ID of the operation
+	ID string `json:"id"`
+	// Origin represents the API discovery engine(s) that discovered this operation
+	Origin []string `json:"origin"`
+	// State represents the state of operation in API Discovery
+	State string `json:"state"`
```

Comment:
Should we provide the set of valid state strings as consts that one can pass here?

Reply:
entirely up to you folks. we do half and half in other places but once we do SDK generation, we will have allowed enums.

Reply:
Have opted to add typed field + consts for these fields https://github.com/cloudflare/cloudflare-go/commit/80d96505236493553fee84065343ea92c44d9d58